app/src/main/cpp/native-lib.cpp

```diff
@@ -36,83 +30,175 @@ namespace
-
-    sdk->oep->load_effect(effect_name);
+    auto oep = reinterpret_cast<BanubaSdkManager*>(jsdk)->oep;
+    //pixel_buffer_sptr s;
+    auto* input_image_data = static_cast<uint8_t*>(env->GetDirectBufferAddress(jimage));
+    auto width = static_cast<int32_t>(jwidth);
+    auto height = static_cast<int32_t>(jheight);
+
+    // create an image
+    using ns_pb = bnb::oep::interfaces::pixel_buffer;
+    int32_t stride = width * 4;
+    ns_pb::plane_sptr rgba_plane_data(input_image_data, [](uint8_t*) { /* DO NOTHING */ });
+    ns_pb::plane_data rgba_plane{rgba_plane_data, static_cast<size_t>(stride * height), stride};
+    std::vector<ns_pb::plane_data> planes{rgba_plane};
+    auto pb_image = ns_pb::create(planes, bnb::oep::interfaces::image_format::bpc8_rgba, width, height, [](auto* pb) {});
+
+    JavaVM* jvm;
+    env->GetJavaVM(&jvm);
+    jobject this_ref = env->NewGlobalRef(thiz);
+
+    // Callback for received pixel buffer from the offscreen effect player
+    auto get_pixel_buffer_callback = [this_ref, jvm](image_processing_result_sptr result) {
+        if (result != nullptr) {
+            // Callback for update data in render thread
+            auto get_image_callback = [this_ref, jvm](pixel_buffer_sptr image) {
+                if (image == nullptr) {
+                    return;
+                }
+                JNIEnv* env = nullptr;
+
+                // double check it's all ok
+                int getEnvStat = jvm->GetEnv((void**)&env, JNI_VERSION_1_6);
+                if (getEnvStat == JNI_EDETACHED) {
+                    std::cout << "GetEnv: not attached" << std::endl;
+                    if (jvm->AttachCurrentThread((JNIEnv**)&env, nullptr) != JNI_OK) {
+                        std::cout << "Failed to attach" << std::endl;
+                        return;
+                    }
+                } else if (getEnvStat == JNI_EVERSION) {
+                    std::cout << "GetEnv: version not supported" << std::endl;
+                    return;
+                }
+
+                jclass jcallback_class = env->GetObjectClass(this_ref);
+                jmethodID jcallback_method = env->GetMethodID(jcallback_class, "oepDataReady", "([BII)V");
+
+                auto size = image->get_width() * image->get_height() * image->get_bytes_per_pixel();
+                auto byte_array = env->NewByteArray(size);
+                env->SetByteArrayRegion(byte_array, 0, size, reinterpret_cast<const jbyte*>((uint8_t*)image->get_base_sptr().get()));
+
+                // call callback
+                env->CallVoidMethod(this_ref, jcallback_method, byte_array, image->get_width(), image->get_height());
+
+                if (env->ExceptionCheck()) {
+                    env->ExceptionDescribe();
+                }
+                env->DeleteGlobalRef(this_ref);
+
+                jvm->DetachCurrentThread();
+            };
+            // Get texture id from shared context and render it
```

Comment:
misleading comment - comment mentioned some texture, but you are requesting RGBA image as buffer